### PR TITLE
xssstate: init at 20220924

### PR DESCRIPTION
**Pull request by onemoresuza**
###### Description of changes

[Package request](https://github.com/NixOS/nixpkgs/issues/244464#issue-1813501572).

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/X11/xssstate/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+}:
+stdenv.mkDerivation rec {
+  name = "xssstate";
+  #
+  # Use the date of the last commit, since there were bug fixes after the 1.1
+  # release.
+  #
+  version = "20220924";
+  src = fetchgit {
+    url = "https://git.suckless.org/xssstate/";
```
Comment (sikmir) [inline pkgs/tools/X11/xssstate/default.nix:15]:
```suggestion
    url = "https://git.suckless.org/xssstate/";
    rev = "5d8e9b49ce2970f786f1e5aa12bbaae83900453f";
```

File: pkgs/tools/X11/xssstate/default.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+, libX11
+, libXScrnSaver
+}:
+stdenv.mkDerivation rec {
+  name = "xssstate";
+  #
+  # Use the date of the last commit, since there were bug fixes after the 1.1
+  # release.
+  #
+  version = "20220924";
```
Comment (sikmir) [inline pkgs/tools/X11/xssstate/default.nix:13]:
```suggestion
  version = "2022-09-24";
```

**Comment by sikmir** [comment]
Result of `nixpkgs-review pr 244592` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>xssstate</li>
  </ul>
</details>

File: pkgs/tools/X11/xssstate/default.nix (hunk 1 above)
Comment (onemoresuza) [reply pkgs/tools/X11/xssstate/default.nix:15]: Fixed as of e634c0bf6b66030b2238b01878a65a20f28a4e46.

File: pkgs/tools/X11/xssstate/default.nix (hunk 2 above)
Comment (onemoresuza) [reply pkgs/tools/X11/xssstate/default.nix]: Fixed as of 957a4628244278c8e61bde1cfe7eeab84d018386.

**Comment by GiggleSquid** [comment]
Squash commits into xssstate: init at 2022-09-24

**Review by GiggleSquid — CHANGES_REQUESTED** [review]
Squash commits into xssstate: init at 2022-09-24

**Review by GiggleSquid — APPROVED** [review]

File: pkgs/tools/X11/xssstate/default.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
+{ lib
+, stdenv
+, fetchgit
+, libX11
+, libXScrnSaver
+}:
+stdenv.mkDerivation rec {
+  name = "xssstate";
```
Comment (sikmir) [inline pkgs/tools/X11/xssstate/default.nix:8]:
```suggestion
  pname = "xssstate";
```

**Comment by onemoresuza** [comment]
Perhaps the short hash of the latest commit would be preferable to its date, since multiple commits can happen on the same day? If so, how is the `nixpkgs` approach to such versioning?

**Comment by sikmir** [comment]
> Perhaps the short hash of the latest commit as the package version would be preferable to its date, since multiple commits can happen on the same day? If so, how is the `nixpkgs` approach to such versioning?

https://nixos.org/manual/nixpkgs/stable/#sec-package-naming:

> If a package is not a release but a commit from a repository, then the version attribute must be the date of that (fetched) commit. The date must be in "unstable-YYYY-MM-DD" format.

**Comment by onemoresuza** [comment]
> > Perhaps the short hash of the latest commit as the package version would be preferable to its date, since multiple commits can happen on the same day? If so, how is the `nixpkgs` approach to such versioning?
> 
> https://nixos.org/manual/nixpkgs/stable/#sec-package-naming:
> 
> > If a package is not a release but a commit from a repository, then the version attribute must be the date of that (fetched) commit. The date must be in "unstable-YYYY-MM-DD" format.

I see. I'll add the `unstable` prefix to `version`.

**Review by sikmir — APPROVED** [review]

**Review by GiggleSquid — APPROVED** [review]

**Pull request merged**